proxy/ParentYamlConfig.cc

```diff
@@ -0,0 +1,224 @@
+/** @file
+
+  Implementation of Parent Yaml Configuration file parsing.
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include <memory>
+#include <vector>
+
+#include "ParentYamlConfig.h"
+
+ParentYamlConfig::ParentYamlConfig() : dest_size(0) {}
+
+ParentYamlConfig::~ParentYamlConfig() {}
+
+parent_dest &
+operator>>(const YAML::Node &node, parent_dest &p)
+{
+  // get all the strings
+  if (node["dest_domain"]) {
+    p.dest_domain = node["dest_domain"].as<std::string>();
+  }
```

Comment:
Consider instead writing:
```
  auto *np = &node["dest_domain"];
  if (*np) {
    p.dest_domain = np->as<std::string>();
  }
  np = &node["dest_host"];
  if (*np) {
    p.dest_host = np->as<std::string>();
  }
  .
  .
  .
```
If the [] operator call results in calling an external function, the compiler probably won't optimize the two calls into a single call, as seen in this example code:  https://godbolt.org/z/0zzf4q . It doesn't seem to matter that the call is to a const member function.  It's apparently considered too risky to optimize this, presumably because of potential function-side effects.

Reply:
thanks @ywkaras I'll definitely consider the change.  still working on this.

Reply:
Yes, this is an issue with the YAMLCPP API IMHO. I'm trying to recall how I deal with this - I think you couldn't use `if (auto n = node["dest_host"] ; n)` for some reason. Researching that again.

Reply:
@ywkaras and @SolidWallOfCode `if (auto n = node["dest_host"] ; n)` doesn't compile